### PR TITLE
[genie][1.0] add new package for genie gesture engine

### PR DESCRIPTION
**Pull request by raisjn**
genie is a gesture based launcher - it recognizes gestures and runs commands

### CODE REVIEW

File: package/rmkit/package [hunk 1]
```diff
@@ -11,17 +11,46 @@ image=python:v1.1
...
+
+    package() {
+        install -D -m 755 "$srcdir"/src/build/genie "$pkgdir"/opt/bin/genie
+        install -D -m 644 "$srcdir"/genie.service "$pkgdir"/lib/systemd/system/genie.service
+        install -D -m 644 "$srcdir"/src/genie/example.conf "$pkgdir"/opt/etc/genie.example.conf
+    }
+
+    configure() {
+        systemctl daemon-reload
+        systemctl enable genie --now
```
Comment (Eeems) [inline package/rmkit/package]: I thought we were avoiding doing this for users for most things?

File: package/rmkit/package [hunk 2]
```diff
@@ -11,17 +11,46 @@ image=python:v1.1
...
+
+    package() {
+        install -D -m 755 "$srcdir"/src/build/genie "$pkgdir"/opt/bin/genie
+        install -D -m 644 "$srcdir"/genie.service "$pkgdir"/lib/systemd/system/genie.service
+        install -D -m 644 "$srcdir"/src/genie/example.conf "$pkgdir"/opt/etc/genie.example.conf
+    }
+
+    configure() {
+        systemctl daemon-reload
+        echo "Run 'systemctl enable genie --now' to enable genie"
```
Comment (Eeems) [inline package/rmkit/package:41]: I think you can use how-to-enable
```suggestion
       how-to-enable genie
```

**Comment by danshick** [comment]
Just installed, I'll test drive for the next day or two.

**Comment by Eeems** [comment]
I think @PeterGrace also has installed and tested on the rM2.

**Comment by danshick** [comment]
So far genie is working well.

**Review by Eeems — APPROVED** [review]

**Comment by PeterGrace** [comment]
> I think @PeterGrace also has installed and tested on the rM2.

Yes indeed -- this is my configuration for the reMarkable 2, specifically so that the full button features of oxide are retained even though there aren't physicalized buttons on the reMarkable 2.  One item of note: see how the paths on the command have to be fully qualified; I'm not sure if there's a genie package/config issue here but the exec'd command doesn't have /opt/bin in the path.  That could be a different issue (for instance, does the toltec installer include /opt/bin in the shell rc file?  At  any rate, the below config does work but beware: that "1 finger tap" for oxide menu also triggers if, like me, you're a lefty and lay your hand on the screen while writing!

```
gesture=tap
fingers=1
command=/opt/bin/rot apps call leftHeld
duration=3

gesture=tap
fingers=2
command=/opt/bin/rot screen call screenshot
duration=3

gesture=tap
fingers=3
command=/opt/bin/rot apps get applications | /opt/bin/jq '."codes.eeems.erode"' -cr | sed 's|/codes/eeems/oxide1/||' | xargs -I {} /opt/bin/rot --object Application:{} apps call launch
duration=3
```

File: package/rmkit/genie.service [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+Description=Gesture launcher
+After=opt.mount
+StartLimitInterval=30
+StartLimitBurst=5
+
+[Service]
+ExecStart=/opt/bin/genie /opt/etc/genie.conf
+Restart=on-failure
+RestartSec=5
+Environment="HOME=/home/root"
```
Comment (Eeems) [inline package/rmkit/genie.service:14]:
```suggestion
Environment="HOME=/home/root"
Environment="PATH=/opt/bin:/opt/sbin:/opt/usr/bin:/usr/local/bin:/usr/bin:/bin:/usr/local/sbin:/usr/sbin:/sbin"
```

**Comment by raisjn** [comment]
> At any rate, the below config does work but beware: that "1 finger tap" for oxide menu also triggers if, like me, you're a lefty and lay your hand on the screen while writing!

nice report! i guess we can make genie work better by not invoking any gestures if pen is on or near screen so that this doesn't happen as frequently

**Pull request merged**